docs: Precise the abscense of `py: Python` for the #[pyo3(signature)] (#2929)
PyO3/pyo3: guide/src/function/signature.md

    }
}
```

Arguments of type `Python` must not be part of the signature:

```rust

> #[pyfunction]

Reviewer: You must import `pyo3` here.

#[pyo3(signature = (lambda))]
pub fn simple_python_bound_function(

>     py: Python,

Reviewer: You must annotate the lifetime here: `Python<'_>`

    lambda: PyObject,
) -> PyResult<()> {

>     OK()

Reviewer: This must be `Ok(())`

    }
}
```

> Also the following constructs must not be part of the signature::
>  * `py: Python`

Reviewer 2: ```suggestion
Arguments of type `Python` must not be part of the signature:
```

    }
}
```

Arguments of type `Python` must not be part of the signature:

```rust
use pyo3::prelude::*;

#[pyfunction]
#[pyo3(signature = (lambda))]
pub fn simple_python_bound_function(
    py: Python<'_>,
    lambda: PyObject,
) -> PyResult<()> {

>     OK(())

Author: ```suggestion
    Ok(())
```

    }
}
```

Arguments of type `Python` must not be part of the signature:

```rust

> use pyo3::prelude::*;
> 
> #[pyfunction]
> #[pyo3(signature = (lambda))]
> #[allow(dead_code)]

Reviewer: ```suggestion
# #![allow(dead_code)]
# use pyo3::prelude::*;
#[pyfunction]
#[pyo3(signature = (lambda))]
```

You can hide things like imports and allows with a `#` in front.
